### PR TITLE
SDK Release for Recovery Services 2023-02-01

### PR DESCRIPTION
**Pull request by profDisarray**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by profDisarray** [comment]
Can any reviewer help us with the failed checks? We tried the suggested steps, but there's no new file getting update after that

**Comment by profDisarray** [comment]
@ArcturusZhang can you help with the failed checks?

**Comment by profDisarray** [comment]
Hi @fengzhou-msft, @archerzz, @Yao725, @ArthurMa1978 can you help us with the failed validations?

File: sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 ### Other Changes
 
+Changed API version to 2023-02-01.
```
Comment (ArthurMa1978) [inline sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/CHANGELOG.md:13]: Would you like to add some tests?

Comment (ArthurMa1978) [reply sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/CHANGELOG.md:13]: And please run `dotnet build /t:GenerateTests` to update the samples

Comment (profDisarray) [reply sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/CHANGELOG.md:13]: Oh thanks! I was running the command on a different csproj so wasn't getting any change. Ran it on the correct one now.

File: sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs [hunk 2]
```diff
@@ -1721,16 +1723,19 @@ public partial class IaasVmRestoreContent : Azure.ResourceManager.RecoveryServic
...
         public Azure.ResourceManager.RecoveryServicesBackup.Models.VmEncryptionDetails EncryptionDetails { get { throw null; } set { } }
+        public Azure.ResourceManager.Resources.Models.ExtendedLocation ExtendedLocation { get { throw null; } set { } }
         public Azure.ResourceManager.RecoveryServicesBackup.Models.IdentityBasedRestoreDetails IdentityBasedRestoreDetails { get { throw null; } set { } }
         public Azure.ResourceManager.RecoveryServicesBackup.Models.BackupIdentityInfo IdentityInfo { get { throw null; } set { } }
         public bool? OriginalStorageAccountOption { get { throw null; } set { } }
         public string RecoveryPointId { get { throw null; } set { } }
         public Azure.ResourceManager.RecoveryServicesBackup.Models.FileShareRecoveryType? RecoveryType { get { throw null; } set { } }
         public Azure.Core.AzureLocation? Region { get { throw null; } set { } }
         public System.Collections.Generic.IList<int> RestoreDiskLunList { get { throw null; } }
+        public string SecuredVmOSDiskEncryptionSetId { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs:1734]: Could you confirm if the value of this property is always an arm resource id?
If so please change its format to `arm-id`

File: sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs [hunk 3]
```diff
@@ -2588,6 +2594,18 @@ public partial class TargetAfsRestoreInfo
         public string Name { get { throw null; } set { } }
         public Azure.Core.ResourceIdentifier TargetResourceId { get { throw null; } set { } }
     }
+    public enum TargetDiskNetworkAccessOption
+    {
+        SameAsOnSourceDisks = 0,
+        EnablePrivateAccessForAllDisks = 1,
+        EnablePublicAccessForAllDisks = 2,
+    }
+    public partial class TargetDiskNetworkAccessSettings
```
Comment (ArcturusZhang) [inline sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs:2603]: Could we add a prefix to these two types to emphasize its ownership?
Other RPs also have similar things and we require all the classes to have unique names across all SDKs

File: sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs [hunk 4]
```diff
@@ -2588,6 +2594,18 @@ public partial class TargetAfsRestoreInfo
...
+    {
+        public TargetDiskNetworkAccessSettings() { }
+        public string TargetDiskAccessId { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs:2606]: Is the value of this property always an arm resource id?
If so, please change the format of it to `arm-id`

**Comment by ArcturusZhang** [comment]
Please refer to [this document](https://github.com/Azure/autorest.csharp/blob/feature/v3/docs/mgmt/polishing.md#change-the-format-of-a-property) to change the format of properties

File: sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs (hunk 2 above)
Comment (profDisarray) [reply sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs]: updated as per suggestion

File: sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs (hunk 3 above)
Comment (profDisarray) [reply sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs]: updated as per suggestion

File: sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs (hunk 4 above)
Comment (profDisarray) [reply sdk/recoveryservices-backup/Azure.ResourceManager.RecoveryServicesBackup/api/Azure.ResourceManager.RecoveryServicesBackup.netstandard2.0.cs]: updated as per suggestion

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**